src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -437,21 +438,21 @@ string relationshipEntity
         /// LinkingSourceField OR SourceFields are null but relationship between source and linking object exists, no relationship between target and linking object
         /// LinkingTargetField OR TargetField are null but relationship between target and linking object exists, no relationship between source and linking object
         /// </summary>
-        [DataRow(new string[] { "id" }, null, new string[] { "num" }, new string[] { "book_num" }, "SampleEntity1", false, true, false,
+        [DataRow(new string[] { "id" }, null, new string[] { "num" }, null, "SampleEntity1", false, true, false,
             DisplayName = "LinkingSourceField is null, only ForeignKeyPair between LinkingObject and target is present. Invalid Case.")]
-        [DataRow(null, new string[] { "token_id" }, new string[] { "num" }, new string[] { "book_num" }, "SampleEntity1", false, true, false,
+        [DataRow(null, new string[] { "token_id" }, null, new string[] { "book_num" }, "SampleEntity1", false, true, false,
             DisplayName = "SourceField is null, only ForeignKeyPair between LinkingObject and target is present.  Invalid Case.")]
-        [DataRow(new string[] { "id" }, new string[] { "token_id" }, new string[] { "num" }, null, "SampleEntity2", true, false, false,
+        [DataRow(new string[] { "id" }, null, new string[] { "num" }, null, "SampleEntity2", true, false, false,
             DisplayName = "LinkingTargetField is null, only ForeignKeyPair between LinkingObject and source is present. Invalid Case.")]
-        [DataRow(new string[] { "id" }, new string[] { "token_id" }, null, new string[] { "book_num" }, "SampleEntity2", true, false, false,
+        [DataRow(null, new string[] { "token_id" }, null, new string[] { "book_num" }, "SampleEntity2", true, false, false,
             DisplayName = "TargetField is null, , only ForeignKeyPair between LinkingObject and source is present. Invalid Case.")]
-        [DataRow(new string[] { "id" }, null, new string[] { "num" }, new string[] { "book_num" }, "SampleEntity1", true, false, true,
+        [DataRow(new string[] { "id" }, null, new string[] { "num" }, null, "SampleEntity1", true, false, true,
             DisplayName = "LinkingSourceField is null, only ForeignKeyPair between LinkingObject and target is present. Valid Case.")]
-        [DataRow(null, new string[] { "token_id" }, new string[] { "num" }, new string[] { "book_num" }, "SampleEntity1", true, false, true,
+        [DataRow(null, new string[] { "token_id" }, null, new string[] { "book_num" }, "SampleEntity1", true, false, true,
```

Comment:
for this test, your modifications null out what was previously an array with a column name. Does that require any modification to the datarow display names because the tests are now different?

Reply:
These cases are not really valid, where we have only target or source fields as null while the other is non null, because this PR adds validation to ensure that this throws an exception, and we handle that case in the unit tests I added.

I am trying to modify these tests minimally to avoid collisions with that new validation logic.

Reply:
I am still going through the updated tests and trying to get the language in its best form to properly describe the cases, since theyve been modified a bit.